integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDictionaryDecoder.scala

```diff
@@ -175,16 +175,25 @@ case class CarbonDictionaryDecoder(
             val unsafeProjection = UnsafeProjection.create(output.map(_.dataType).toArray)
             var flag = true
             var total = 0L
+
             override final def hasNext: Boolean = iter.hasNext
+
             override final def next(): InternalRow = {
               val startTime = System.currentTimeMillis()
               val row: InternalRow = iter.next()
               val data = row.toSeq(dataTypes).toArray
               dictIndex.foreach { index =>
                 if (data(index) != null) {
-                  data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
-                    .getDictionaryValueForKey(data(index).asInstanceOf[Int]),
-                    getDictionaryColumnIds(index)._3)
+                  try {
+                    data(index) = DataTypeUtil.getDataBasedOnDataType(dicts(index)
+                      .getDictionaryValueForKey(data(index).asInstanceOf[Int]),
+                      getDictionaryColumnIds(index)._3)
+                  }
+                  catch {
+                    case x: ClassCastException => {
```

Comment:
How just exception logging could solve the issue? It can create some other issue, please try to avoid classcast exception first.